### DISCUSSION TITLE
How to classify a simple list of entities?

### DISCUSSION THREAD

**Discussion by datawranglingey** [explosion/spaCy#7908] [Help: Coding & Implementations] [answered]
I am trying to input a list of entity names and use spaCy to help classify those as likely individuals or not likely individuals. Could someone point me in the right direction on how to do this? I assume the model would have to consider each cell as one token. The input would be a simple .csv file with one column that contains the names of entities. 

In the past, I have manually added rules in SQL to spot things like LLC, Inc, Dr, etc. I was wondering if spaCy had libraries or logics that browsed through more comprehensive lists of indicators like these.

Example input: 
- John Smith
- McDonald's
- Company, Inc.
- Gray LLC
- Jane Williams
- Dr. Pepper Corp
- Waste Management of the Municipality of Town

Example Output:
- John Smith - Individual
- McDonald's - Entity
- Company, Inc. - Entity
- Gray LLC - Entity
- Jane Williams - Individual
- Dr. Pepper Corp - Entity
- Waste Management of the Municipality of Town - Entity

**Comment by polm** [comment, marked as answer]
This is not quite something spaCy can do out of the box. The pretrained models can recognize entities in context and give them labels like Person or Organization, but the models are trained on complete sentences, not isolated text. I guess you could make a textcat model but you'd need training data.

Can you clarify why you need to classify just the isolated text? In some cases this problem has no right answer, such as when a company name is also a personal name, like "Sears" or "Margaret O'Leary" or "Calvin Klein", and you could only tell if it's a person or organization in context.

> I assume the model would have to consider each cell as one token. 

Not sure what you mean by a "cell" - do you mean one of your examples? - but there's no reason in any model, spaCy or not, it would have to be one token.

> In the past, I have manually added rules in SQL to spot things like LLC, Inc, Dr, etc. I was wondering if spaCy had libraries or logics that browsed through more comprehensive lists of indicators like these.

I am not aware of anything like that in [the universe](https://spacy.io/universe), but it's possible I missed something.

**Reply by datawranglingey** [reply]
Hi Paul, thank you for your help. I only have the names of vendors along with a few other columns such as Industry_Sector in a vendors spreadsheet. I need to know which vendors are legal entities and which are individuals who would require more protection when handling their personal information. By "cell" I am referring to one cell in a spreadsheet that contains the full entity name. 

I have enough data to train and test a model. I also have a field in the spreadsheet that has the correct classification of the majority of vendors as individuals or legal entities so I could pass these through for the training and testing. 

Any tips on setting or configurations to get right when training the model?

**Reply by polm** [reply]
Thanks for the details, that makes a lot of sense. In that case I recommend you try using the [training quickstart](https://spacy.io/usage/training#quickstart) for a textcat model with exclusive categories. 

Since you have other columns, I would also try including them some way, maybe like this:

```
COLUMN1: col1 value
COLUMN2: col2 value
```

You might also want to have a look at non-spaCy classifiers for "tabular data", though usually those involve more numeric features, which doesn't sound like your data. On the other hand it's possible someone else with the same problem tried to use them before for a problem like yours and you can find some pointers.
